File: Hash Tables/BinaryTreeInorderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        stack = []
        values = []
        
        if root == None:
            return []
        
        stack.append(root)
        while(len(stack)!=0):
            while root!= None:
                stack.append(root)
                root = root.left
            root = stack.pop()
            values.append(root.val)
            root = root.right
            
        return values[:-1]

File: Hash Tables/BinaryTreeInorderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        stack<TreeNode*> nodeStack;
        vector<int> inorderVec;
        
        if (root == nullptr)
            return {};
            
        TreeNode* curNode = root;
        while(curNode!= nullptr || !nodeStack.empty()){
            while (curNode!= nullptr){
                nodeStack.push(curNode);
                curNode = curNode->left;
            }
            curNode = nodeStack.top();
            nodeStack.pop();
            inorderVec.push_back(curNode->val);
            curNode = curNode->right;
        }
        return inorderVec;
    }
};
